### PR TITLE
Update to Quarkus 2.13.7.Final

### LINKED ISSUE #12663: CWE-295 - Improper Certificate Validation vulnerability in io.netty:netty-handler
### Describe the bug

[io.netty:netty-handler](https://github.com/netty/netty.git/netty-handler) is a library that provides an asynchronous event-driven network application framework and tools for rapid development of maintainable high performance and high scalability protocol servers and clients. In other words, Netty is a NIO client server framework which enables quick and easy development of network applications such as protocol servers and clients. It greatly simplifies and streamlines network programming such as TCP and UDP socket server.

Affected versions of this package are vulnerable to Improper Certificate Validation. Certificate hostname validation is disabled by default in `Netty 4.1.x` which makes it potentially susceptible to Man-in-the-Middle attacks.

## Version

<= 18.0.0

## Impact

- Pending  

## Remediation

There is no fixed version for `io.netty:netty-handler`. Possible mitigation was suggested here [https://github.com/netty/netty/issues/8537#issuecomment-822441268](https://github.com/netty/netty/issues/8537#issuecomment-822441268)

## References

-   [GitHub Issue 1](https://github.com/netty/netty/issues/10806)
-   [GitHub Issue 2](https://github.com/netty/netty/issues/8537)
-   [GitHub Issue 3](https://github.com/netty/netty/issues/9930)
-   [GitHub Issue 4](https://github.com/netty/netty/issues/10362)

## Credits

-   [https://security.snyk.io/vuln/SNYK-JAVA-IONETTY-1042268](https://security.snyk.io/vuln/SNYK-JAVA-IONETTY-1042268)

## Additional information

Netty Handler is a transitive dependency coming from Quarkus. According to the Netty team, the fix should be available on Netty 5

### LINKED ISSUE #16119: [CVE-2022-41881] Denial of Service (DoS) vulnerability in io.netty:netty-codec-haproxy
### Detailed paths

-   _Introduced through_: org.keycloak:keycloak-quarkus-server-app@999-SNAPSHOT › org.keycloak:keycloak-quarkus-server@999-SNAPSHOT › io.quarkus:[quarkus-vertx@2.13.5.Final](mailto:quarkus-vertx@2.13.5.Final) › io.netty:[netty-codec-haproxy@4.1.82.Final](mailto:netty-codec-haproxy@4.1.82.Final)

# Overview

Affected versions of this package are vulnerable to Denial of Service (DoS) due to a stack overflow that can be triggered by a malformed message containing a deeply nested TLV. (The only limitation on such recursion is that the TLV length cannot exceed `0xffff`.)

**NOTE:** The StackOverflowError is caught if HAProxyMessageDecoder is used as part of Netty’s ChannelPipeline.

# Details

Denial of Service (DoS) describes a family of attacks, all aimed at making a system inaccessible to its intended and legitimate users.

Unlike other vulnerabilities, DoS attacks usually do not aim at breaching security. Rather, they are focused on making websites and services unavailable to genuine users resulting in downtime.

One popular Denial of Service vulnerability is DDoS (a Distributed Denial of Service), an attack that attempts to clog network pipes to the system by generating a large volume of traffic from many machines.

When it comes to open source libraries, DoS vulnerabilities allow attackers to trigger such a crash or crippling of the service by using a flaw either in the application code or from the use of open source libraries.

# Remediation

Upgrade io.netty:netty-codec-haproxy to version 4.1.86.Final or higher.

# References

-   [GitHub Commit](https://github.com/netty/netty/commit/cd91cf3c99123bd1e53fd6a1de0e3d1922f05bb2)
- [CVE-2022-41881](https://www.cve.org/CVERecord?id=CVE-2022-41881)
- https://github.com/netty/netty/security/advisories/GHSA-fx2c-96vj-985v
- Related to #16120

### LINKED ISSUE #16120: [CVE-2022-41915] HTTP Response Splitting vulnerability in io.netty:netty-codec
### Description

-   Package Manager: maven
-   Vulnerable module: io.netty:netty-codec
-   Introduced through: org.keycloak:keycloak-quarkus-server-app@999-SNAPSHOT, org.keycloak:keycloak-quarkus-server@999-SNAPSHOT and others

### Detailed paths

-   _Introduced through_: org.keycloak:keycloak-quarkus-server-app@999-SNAPSHOT › org.keycloak:keycloak-quarkus-server@999-SNAPSHOT › io.quarkus:[quarkus-vertx@2.13.5.Final](mailto:quarkus-vertx@2.13.5.Final) › io.quarkus:[quarkus-netty@2.13.5.Final](mailto:quarkus-netty@2.13.5.Final) › io.netty:[netty-codec@4.1.82.Final](mailto:netty-codec@4.1.82.Final)

# Overview

[io.netty:netty-codec](https://github.com/netty/netty) is an event-driven asynchronous network application framework.

Affected versions of this package are vulnerable to HTTP Response Splitting when calling `DefaultHttpHeaders.set` on an iterator of values, because header value validation is not performed.

# Workaround

This vulnerability can be worked around by changing the `DefaultHttpHeaders.set(CharSequence, Iterator<?>)` call into a `remove()` call, and calling `add()` in a loop over the iterator.

# Remediation

Upgrade `io.netty:netty-codec` to version 4.1.86.Final or higher.

# References

-   [GitHub Commit](https://github.com/netty/netty/commit/fe18adff1c2b333acb135ab779a3b9ba3295a1c4)
- [CVE-2022-41915](https://www.cve.org/CVERecord?id=CVE-2022-41915)

### LINKED ISSUE #16222: operator doesn't watch other namespaces
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

operator

### Describe the bug

according to [this](https://github.com/keycloak/keycloak/pull/12105) and [this](https://github.com/keycloak/keycloak/issues/11938#issuecomment-1125201574) the operator should be watching namespaces defined in the `QUARKUS_OPERATOR_SDK_NAMESPACES` env variable, but it doesn't seem to work for me.

### Version

20.0.2

### Expected behavior

reconcile keycloak resources in other namespaces than the operator runs in.

### Actual behavior

reconciles only in the operator's namespace

### How to Reproduce?

deploy operator with:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2022-12-20 - 19:05:31 +0000
    deployment.kubernetes.io/revision: "2"
    meta.helm.sh/release-name: keycloak-operator
    meta.helm.sh/release-namespace: keycloak-operator
  labels:
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: keycloak-operator
    app.kubernetes.io/version: 20.0.2
  name: keycloak-operator
  namespace: keycloak-operator
spec:
  progressDeadlineSeconds: 600
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app.kubernetes.io/name: keycloak-operator
      app.kubernetes.io/version: 20.0.2
  strategy:
    rollingUpdate:
      maxSurge: 25%
      maxUnavailable: 25%
    type: RollingUpdate
  template:
    metadata:
      annotations:
        app.quarkus.io/build-timestamp: 2022-12-20 - 19:05:31 +0000
      creationTimestamp: null
      labels:
        app.kubernetes.io/name: keycloak-operator
        app.kubernetes.io/version: 20.0.2
    spec:
      containers:
      - env:
        - name: OPERATOR_KEYCLOAK_IMAGE
          value: quay.io/keycloak/keycloak:20.0.2
        - name: QUARKUS_OPERATOR_SDK_NAMESPACES
          value: iam-test
        image: quay.io/keycloak/keycloak-operator:20.0.2
        imagePullPolicy: Always
        livenessProbe:
          failureThreshold: 3
          httpGet:
            path: /q/health/live
            port: 8080
            scheme: HTTP
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 10
        name: keycloak-operator
        ports:
        - containerPort: 8080
          name: http
          protocol: TCP
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /q/health/ready
            port: 8080
            scheme: HTTP
          periodSeconds: 30
          successThreshold: 1
          timeoutSeconds: 10
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: keycloak-operator
      serviceAccountName: keycloak-operator
      terminationGracePeriodSeconds: 30

```

create a keycloak resource in the iam-test namespace:
```
apiVersion: k8s.keycloak.org/v2alpha1
kind: Keycloak
metadata:
  annotations:
    meta.helm.sh/release-name: iam-test
    meta.helm.sh/release-namespace: iam-test
  creationTimestamp: "2023-01-03T12:54:17Z"
  generation: 1
  labels:
    app.kubernetes.io/managed-by: Helm
  name: internal-test
  namespace: iam-test
  resourceVersion: "844051854"
  uid: ff5a05cb-8651-49fe-84fe-9ce437b0d6b4
spec:
  db:
    host: iam-db
    passwordSecret:
      key: password
      name: iam-db.credentials.postgresql.acid.zalan.do
    usernameSecret:
      key: username
      name: iam-db.credentials.postgresql.acid.zalan.do
    vendor: postgres
  hostname:
    hostname: login-internal-test.example.net
  http:
    httpEnabled: true
    httpPort: 8080
    httpsPort: 8443
    tlsSecret: keycloak-tls
  image: repo/keycloak-customized:20.0.2
  ingress:
    enabled: false
  instances: 1

```

### Anything else?

the operator's logs are also indicating that the watched namespace is only `keycloak-operator`:

```
keycloak-operator-65fcbbf475-kmn5g keycloak-operator __  ____  __  _____   ___  __ ____  ______
keycloak-operator-65fcbbf475-kmn5g keycloak-operator  --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
keycloak-operator-65fcbbf475-kmn5g keycloak-operator  -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
keycloak-operator-65fcbbf475-kmn5g keycloak-operator --\___\_\____/_/ |_/_/|_/_/|_|\____/___/
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:31,573 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.operator-sdk.generate-csv" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,172 INFO  [io.qua.ope.run.OperatorProducer] (main) Quarkus Java Operator SDK extension 4.0.4 (commit: d1a3d1e on branch: d1a3d1ef9b22414b740e9f4c0d74d62a673031e3) built on Fri Oct 21 09:02:30 UTC 2022
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,238 INFO  [io.jav.ope.Operator] (main) Registered reconciler: 'keycloakcontroller' for resource: 'class org.keycloak.operator.crds.v2alpha1.deployment.Keycloak' for namespace(s): [keycloak-operator]
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,241 INFO  [io.jav.ope.Operator] (main) Registered reconciler: 'keycloakrealmimportcontroller' for resource: 'class org.keycloak.operator.crds.v2alpha1.realmimport.KeycloakRealmImport' for namespace(s): [keycloak-operator]
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,242 INFO  [io.qua.ope.run.AppEventListener] (main) Starting operator.
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,242 INFO  [io.jav.ope.Operator] (main) Operator SDK 3.2.4 (commit: dfae8d5) built on Fri Oct 21 08:03:55 UTC 2022 starting...
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,242 INFO  [io.jav.ope.Operator] (main) Client version: 5.12.4
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,249 INFO  [io.jav.ope.pro.Controller] (main) Starting 'keycloakcontroller' controller for reconciler: org.keycloak.operator.controllers.KeycloakController_ClientProxy, resource: org.keycloak.operator.crds.v2alpha1.deployment.Keycloak
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:33,297 WARN  [io.fab.kub.cli.int.VersionUsageUtils] (main) The client is using resource type 'keycloaks' with unstable version 'v2alpha1'
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:34,952 INFO  [io.jav.ope.pro.Controller] (main) 'keycloakcontroller' controller started, pending event sources initialization
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:34,953 INFO  [io.jav.ope.pro.Controller] (main) Starting 'keycloakrealmimportcontroller' controller for reconciler: org.keycloak.operator.controllers.KeycloakRealmImportController_ClientProxy, resource: org.keycloak.operator.crds.v2alpha1.realmimport.KeycloakRealmImport
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:34,955 WARN  [io.fab.kub.cli.int.VersionUsageUtils] (main) The client is using resource type 'keycloakrealmimports' with unstable version 'v2alpha1'
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:35,016 INFO  [io.jav.ope.pro.Controller] (main) 'keycloakrealmimportcontroller' controller started, pending event sources initialization
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:35,202 INFO  [io.quarkus] (main) keycloak-operator 20.0.2 on JVM (powered by Quarkus 2.13.3.Final) started in 4.032s. Listening on: http://0.0.0.0:8080
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:35,202 INFO  [io.quarkus] (main) Profile prod activated.
keycloak-operator-65fcbbf475-kmn5g keycloak-operator 2023-01-03 12:43:35,203 INFO  [io.quarkus] (main) Installed features: [cdi, kubernetes, kubernetes-client, openshift-client, operator-sdk, rest-client, rest-client-jackson, resteasy-jackson, smallrye-context-propagation, smallrye-health, vertx]

```

tested the versions backwards, and 19.0.2 seems to work as expected:

```
keycloak-operator-78d7dcc795-fbvcs keycloak-operator 2023-01-03 14:55:10,054 INFO  [io.jav.ope.Operator] (main) Registered reconciler: 'keycloakcontroller' for resource: 'class org.keycloak.operator.crds.v2alpha1.deployment.Keycloak' for namespace(s): [iam-test]
keycloak-operator-78d7dcc795-fbvcs keycloak-operator 2023-01-03 14:55:10,062 INFO  [io.jav.ope.Operator] (main) Registered reconciler: 'keycloakrealmimportcontroller' for resource: 'class org.keycloak.operator.crds.v2alpha1.realmimport.KeycloakRealmImport' for namespace(s): [iam-test]
```

### LINKED ISSUE #16755: Update to Quarkus 2.13.7.Final
### Description

Quarkus team released 2.13.7.Final. It is important to upgrade to fix a coupe of CVEs on transitive dependencies, plus a couple of bug fixes present on this release.

### PR DESCRIPTION
**Pull request by abstractj**
Closes #16755, Closes #16120, Closes #16119, Closes #12663, Closes #16222

### CODE REVIEW

**Comment by stianst** [comment]
@abstractj tests didn't run because I forgot to add pom.xml files to conditions for workflows, added now https://github.com/keycloak/keycloak/commit/d81794123bb850de688dda9ca29d65caf819f23c. Mark this as "ready" as otherwise I wasn't able to update it.

**Comment by pedroigor** [comment]
@abstractj Jackson version is different. See https://github.com/quarkusio/quarkus/blob/2.13.7.Final/bom/application/pom.xml#L86.

If you are using `set-quarkus-version.sh`, the version is not set properly because the name of our property is different than what Quarkus is using. I can fix this.

**Comment by abstractj** [comment]
@pedroigor thank you! I did the update to this PR, and also checked Quarkus dependencies. But if I missed something, I'd appreciate your help.

**Comment by abstractj** [comment]
@keycloak/maintainers  it would be fantastic if we manage to get this merged right before the release. Plus, if approved, I can backport those changes to `release/20.0` branch. Including it, will fix the CVEs present on Netty.

**Review by pedroigor — DISMISSED** [review]

**Review by pedroigor — DISMISSED** [review]
@vmuzikar Does it look good for you?

File: operator/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <quarkus.operator.sdk.version>4.0.6</quarkus.operator.sdk.version>
-        <quarkus.version>2.13.6.Final</quarkus.version>
```
Comment (vmuzikar) [inline operator/pom.xml:36]: The Quarkus version for the Operator is tied to the SDK, we cannot override the version like this, I'm afraid. This property is here just to align some additional dependencies with the SDK Quarkus version.

**Review by vmuzikar — CHANGES_REQUESTED** [review]

File: operator/pom.xml (hunk 1 above)
Comment (abstractj) [reply operator/pom.xml:36]: @vmuzikar updated the Operator SDK to keep it in sync with Quarkus. If we don't upgrade to the next  release, I'm afraid that we won't be free of several CVEs currently reported.

**Review by pedroigor — APPROVED** [review]

**Review by vmuzikar — APPROVED** [review]
LGTM, thanks.

**Pull request merged**